src/lpython/parser/parser.yy

```diff
@@ -488,13 +488,13 @@ if_statement
     ;
 
 for_statement
-    : KW_FOR tuple_item KW_IN expr ":" sep statements {
+    : KW_FOR id_item KW_IN expr ":" sep statements {
         $$ = FOR_01($2, $4, $7, @$); }
-    | KW_FOR tuple_item KW_IN expr ":" sep statements KW_ELSE ":"
+    | KW_FOR id_item KW_IN expr ":" sep statements KW_ELSE ":"
         sep statements { $$ = FOR_02($2, $4, $7, $11, @$); }
-    | KW_FOR tuple_item KW_IN expr ":" TK_TYPE_COMMENT TK_NEWLINE statements {
+    | KW_FOR id_item KW_IN expr ":" TK_TYPE_COMMENT TK_NEWLINE statements {
         $$ = FOR_03($2, $4, $6, $8, @$); }
-    | KW_FOR tuple_item KW_IN expr ":" TK_TYPE_COMMENT TK_NEWLINE statements
+    | KW_FOR id_item KW_IN expr ":" TK_TYPE_COMMENT TK_NEWLINE statements
```

Comment:
Just wanted to add, thanks @akshanshbhatt for your discussion and explanation [here](https://github.com/lcompilers/lpython/pull/709#issuecomment-1169704750). It was helpful with this commit, not sure if it's the acceptable solution but I still appreciate the verbosity.

Also, cc: @certik on asking the right question there :) It helps others in the future PRs.